### PR TITLE
[Components with more than 1 service] LPS-183929 Refactor Document Library components

### PR DESCRIPTION
**Pull request by RenanUmV**
[LPS-183929](https://issues.liferay.com/browse/LPS-183929)

### CODE REVIEW

**Comment by RenanUmV** [comment]
ci:test:document

**Comment by RenanUmV** [comment]
ci:test:dl-store

**Review by rafaprax — APPROVED** [review]

File: modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/repository/cache/PortalCacheRegistryItemImpl.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+		for (LiferayDynamicCapability liferayDynamicCapability :
+				_liferayDynamicCapabilities.values()) {
+
+			liferayDynamicCapability.clear();
+		}
+
+		_liferayDynamicCapabilities.clear();
+	}
+
+	private final Map<DocumentRepository, LiferayDynamicCapability>
```
Comment (dantewang) [inline modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/repository/cache/PortalCacheRegistryItemImpl.java:54]: @RenanUmV @rafaprax After extracting this class, this map is never populated; on the other hand, the original map in `PortalCapabilityLocatorImpl` will not be cleared.

Comment (dantewang) [reply modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/repository/cache/PortalCacheRegistryItemImpl.java:54]: @zhangyan0701 Do we have any test related to `PortalCapabilityLocator`?

**Comment by dantewang** [comment]
@RenanUmV Please fix the issue mentioned above and resend, thanks!

When you resend, please use a new branch, thanks!

File: modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/repository/cache/PortalCacheRegistryItemImpl.java (hunk 1 above)
Comment (zhangyan0701) [reply modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/repository/cache/PortalCacheRegistryItemImpl.java:54]: > @zhangyan0701 Do we have any test related to `PortalCapabilityLocator`?

Hi @dantewang What does `PortalCapabilityLocator` do? Are there manual steps that may trigger this class? I am not really familiar with DM repository.

**Pull request closed without merging**